make-indices.py

```diff
@@ -83,10 +83,88 @@ def make_cluster_index(sizes):
     np.save('gmsc-db/GMSC10.cluster.index.npy', ix)
     np.save('gmsc-db/GMSC10.cluster.data.npy', data)
 
+@TaskGenerator
+def create_tax_index(infile,outfile1,outfile2):
+    import lzma
+    tax_set = set()
+    n = 0
+    with lzma.open(infile,'rt') as f:
+        for line in f:
+            linelist = line.strip().split('\t')
+            if len(linelist) == 2:
+                tax_set.add(linelist[1])
+            else:
+                tax_set.add('Unknown')
+            
+    tax_order = sorted(list(tax_set))
+    with open(outfile1,'wt') as out:
+        for item in tax_order:
+            out.write(f'{n}\t{item}\n')
+            n += 1
+    
+    tax_dict = {}
+    with open(outfile1,'rt') as f:
+        for line in f:
+            index,tax = line.strip().split('\t')
+            tax_dict[tax] = index
+
+    with open(outfile2,'wt') as out:
+        with lzma.open(infile,'rt') as f:
+            for line in f:
+                linelist = line.strip().split('\t')
+                if len(linelist) == 2:
+                    out.write(f'{linelist[0]}\t{tax_dict[linelist[1]]}\n')
+                else:
+                    out.write(f'{linelist[0]}\t0\n')
+
+@TaskGenerator
+def create_habitat_index(infile,outfile1,outfile2):
+    import lzma
+    habitat_set = set()
+    n = 0
+    with lzma.open(infile,'rt') as f:
+        for line in f:
+            smorf,habitat = line.strip().split('\t')
+            habitat_set.add(habitat)
+    habitat_order = sorted(list(habitat_set))
+    with open(outfile1,'wt') as out:
+        for item in habitat_order:
+            out.write(f'{n}\t{item}\n')
+            n += 1
+
+    habitat_dict = {}
+    with open(outfile1,'rt') as f:
+        for line in f:
```

Comment:
This can be created in the same loop as you write out the `outfile1` instead of reading it back in, no?